File: buybeats/src/main/java/com/ibm/buybeats/service/OrderService.java
package com.ibm.buybeats.service;

import java.util.List;

import com.ibm.buybeats.entity.Address;
import com.ibm.buybeats.entity.CardDetails;
import com.ibm.buybeats.entity.Cart;
import com.ibm.buybeats.entity.Order;
import com.ibm.buybeats.entity.OrderDetails;
import com.ibm.buybeats.entity.User;
import com.ibm.buybeats.exception.NoAddressFound;
import com.ibm.buybeats.exception.NoCardsFound;
import com.ibm.buybeats.exception.NoOrderFoundException;
import com.ibm.buybeats.exception.StockNotAvaialble;

/**
 * This interface has services for Order
 * @author Arya P Menon
 * @author Darshan Kansara
 * @author Neha
 * @version 1.0
 */

public interface OrderService {

	List<Order> showOrders(int oid) throws NoOrderFoundException;

	Order placeOrder(int uid, int aid) throws StockNotAvaialble;
	
	List<OrderDetails> viewOrderDetails(int oid);
	
	List<Address> viewAddress(int uid) throws NoAddressFound;
	
	List<CardDetails> viewCard(int uid) throws NoCardsFound;

	String confirmOrder(int oid, int code);
}

File: buybeats-react-app/src/Services/OrderService.js
import axios from "axios";

export default class OrderService {
    constructor() {
        this.uri = "http://localhost:8870/order" ;
        this.addresses = [];
    }


    getAddresses() {
        const uid =JSON.parse(localStorage.getItem("user")).uid;
        return axios.get(this.uri+"/viewAddress/"+uid);
    }

    confirmation(oid,code){
        
        return axios.get(this.uri+"/confirmOrder/"+oid+"?code="+code);
    }

    
}